<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditIndexDialog</class>
 <widget class="QDialog" name="EditIndexDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>686</width>
    <height>543</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Edit Index Schema</string>
  </property>
  <property name="windowIcon">
   <iconset resource="icons/icons.qrc">
    <normaloff>:/icons/index_create</normaloff>:/icons/index_create</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <item>
    <layout class="QFormLayout" name="formLayout">
     <item row="0" column="0">
      <widget class="QLabel" name="labelIndexName">
       <property name="text">
        <string>&amp;Name</string>
       </property>
       <property name="buddy">
        <cstring>editIndexName</cstring>
       </property>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QLineEdit" name="editIndexName"/>
     </item>
     <item row="1" column="0">
      <widget class="QLabel" name="labelTableName">
       <property name="text">
        <string>&amp;Table</string>
       </property>
       <property name="buddy">
        <cstring>comboTableName</cstring>
       </property>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QComboBox" name="comboTableName"/>
     </item>
     <item row="2" column="0">
      <widget class="QLabel" name="labelIndexUnique">
       <property name="text">
        <string>&amp;Unique</string>
       </property>
       <property name="buddy">
        <cstring>checkIndexUnique</cstring>
       </property>
      </widget>
     </item>
     <item row="2" column="1">
      <widget class="QCheckBox" name="checkIndexUnique">
       <property name="text">
        <string/>
       </property>
      </widget>
     </item>
     <item row="4" column="0">
      <widget class="QLabel" name="labelIndexColumns">
       <property name="text">
        <string>Colu&amp;mns</string>
       </property>
       <property name="buddy">
        <cstring>tableIndexColumns</cstring>
       </property>
      </widget>
     </item>
     <item row="4" column="1">
      <widget class="QSplitter" name="splitter">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <widget class="QWidget" name="layoutWidget">
        <layout class="QHBoxLayout" name="horizontalLayout">
         <item>
          <widget class="QTableWidget" name="tableTableColumns">
           <property name="editTriggers">
            <set>QAbstractItemView::NoEditTriggers</set>
           </property>
           <property name="showDropIndicator" stdset="0">
            <bool>false</bool>
           </property>
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::SingleSelection</enum>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Table column</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Type</string>
            </property>
           </column>
          </widget>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout">
           <item>
            <spacer name="verticalSpacer">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QToolButton" name="buttonToIndex">
             <property name="icon">
              <iconset resource="icons/icons.qrc">
               <normaloff>:/icons/run</normaloff>:/icons/run</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="buttonFromIndex">
             <property name="icon">
              <iconset resource="icons/icons.qrc">
               <normaloff>:/icons/resultset_previous.png</normaloff>:/icons/resultset_previous.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="verticalSpacer_2">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>40</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QTableWidget" name="tableIndexColumns">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="MinimumExpanding">
             <horstretch>0</horstretch>
             <verstretch>1</verstretch>
            </sizepolicy>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>250</height>
            </size>
           </property>
           <property name="editTriggers">
            <set>QAbstractItemView::NoEditTriggers</set>
           </property>
           <property name="showDropIndicator" stdset="0">
            <bool>false</bool>
           </property>
           <property name="alternatingRowColors">
            <bool>true</bool>
           </property>
           <property name="selectionMode">
            <enum>QAbstractItemView::SingleSelection</enum>
           </property>
           <property name="selectionBehavior">
            <enum>QAbstractItemView::SelectRows</enum>
           </property>
           <attribute name="horizontalHeaderStretchLastSection">
            <bool>true</bool>
           </attribute>
           <attribute name="verticalHeaderVisible">
            <bool>false</bool>
           </attribute>
           <column>
            <property name="text">
             <string>Index column</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Order</string>
            </property>
           </column>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="SqlTextEdit" name="sqlTextEdit" native="true">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>100</height>
         </size>
        </property>
        <property name="readOnly" stdset="0">
         <bool>true</bool>
        </property>
       </widget>
      </widget>
     </item>
     <item row="3" column="0">
      <widget class="QLabel" name="label">
       <property name="toolTip">
        <string>For restricting the index to only a part of the table you can specify a WHERE clause here that selects the part of the table that should be indexed</string>
       </property>
       <property name="text">
        <string>Partial inde&amp;x clause</string>
       </property>
       <property name="buddy">
        <cstring>editPartialClause</cstring>
       </property>
      </widget>
     </item>
     <item row="3" column="1">
      <widget class="QLineEdit" name="editPartialClause"/>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SqlTextEdit</class>
   <extends>QWidget</extends>
   <header>sqltextedit.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>editIndexName</tabstop>
  <tabstop>comboTableName</tabstop>
  <tabstop>checkIndexUnique</tabstop>
  <tabstop>editPartialClause</tabstop>
  <tabstop>tableTableColumns</tabstop>
  <tabstop>tableIndexColumns</tabstop>
  <tabstop>buttonToIndex</tabstop>
  <tabstop>buttonFromIndex</tabstop>
 </tabstops>
 <resources>
  <include location="icons/icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>258</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>326</x>
     <y>499</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>comboTableName</sender>
   <signal>currentIndexChanged(QString)</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>tableChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>74</y>
    </hint>
    <hint type="destinationlabel">
     <x>236</x>
     <y>31</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editIndexName</sender>
   <signal>textChanged(QString)</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>checkInput()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>429</x>
     <y>14</y>
    </hint>
    <hint type="destinationlabel">
     <x>443</x>
     <y>39</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableIndexColumns</sender>
   <signal>cellChanged(int,int)</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>checkInput()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>443</x>
     <y>170</y>
    </hint>
    <hint type="destinationlabel">
     <x>566</x>
     <y>40</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>checkIndexUnique</sender>
   <signal>toggled(bool)</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>checkInput()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>153</x>
     <y>100</y>
    </hint>
    <hint type="destinationlabel">
     <x>304</x>
     <y>251</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonToIndex</sender>
   <signal>clicked()</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>addToIndex()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>417</x>
     <y>248</y>
    </hint>
    <hint type="destinationlabel">
     <x>385</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonFromIndex</sender>
   <signal>clicked()</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>removeFromIndex()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>417</x>
     <y>286</y>
    </hint>
    <hint type="destinationlabel">
     <x>350</x>
     <y>95</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableIndexColumns</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>removeFromIndex(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>520</x>
     <y>236</y>
    </hint>
    <hint type="destinationlabel">
     <x>684</x>
     <y>175</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>tableTableColumns</sender>
   <signal>doubleClicked(QModelIndex)</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>addToIndex(QModelIndex)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>231</x>
     <y>357</y>
    </hint>
    <hint type="destinationlabel">
     <x>19</x>
     <y>205</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editPartialClause</sender>
   <signal>textChanged(QString)</signal>
   <receiver>EditIndexDialog</receiver>
   <slot>checkInput()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>242</x>
     <y>129</y>
    </hint>
    <hint type="destinationlabel">
     <x>47</x>
     <y>103</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>tableChanged(QString)</slot>
  <slot>checkInput()</slot>
  <slot>addToIndex()</slot>
  <slot>addToIndex(QModelIndex)</slot>
  <slot>removeFromIndex()</slot>
  <slot>removeFromIndex(QModelIndex)</slot>
 </slots>
</ui>
